lib/components/AlbumScreen/speed_menu.dart

```diff
@@ -0,0 +1,152 @@
+
+  final Color iconColor;
+  final Function()? scrollFunction;
+
+  @override
+  State<SpeedMenu> createState() => _SpeedMenuState();
+}
+
+class _SpeedMenuState extends State<SpeedMenu> {
+  final _textController = TextEditingController(
+      text: FinampSettingsHelper.finampSettings.playbackSpeed.toString());
+  final _formKey = GlobalKey<FormState>();
+
+  var currentSpeed = FinampSettingsHelper.finampSettings.playbackSpeed;
+
+  InputDecoration inputFieldDecoration() {
+    return InputDecoration(
+      filled: true,
+      fillColor: widget.iconColor.withOpacity(0.1),
+      contentPadding:
+          const EdgeInsets.symmetric(horizontal: 16.0, vertical: 4.0),
+      label: Center(child: Text(AppLocalizations.of(context)!.speed)),
+      floatingLabelBehavior: FloatingLabelBehavior.never,
+      border: OutlineInputBorder(
+        borderSide: BorderSide.none,
+        borderRadius: BorderRadius.circular(16),
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      decoration: BoxDecoration(
+        borderRadius: BorderRadius.circular(10),
+        color: widget.iconColor.withOpacity(0.1),
+      ),
+      margin: EdgeInsets.only(top: 10.0, bottom: 10.0, left: 10.0, right: 10.0),
+      child: Padding(
+        padding: const EdgeInsets.only(top: 12.0, bottom: 12.0),
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: [
+            Padding(
+                padding: EdgeInsets.symmetric(horizontal: 12.0),
+                child: PresetChips(
+                    type: 'speed',
+                    mainColour: widget.iconColor,
+                    values: presets,
+                    activeValue: currentSpeed,
+                    onPressed: () {
+                      setState(() {
+                        currentSpeed =
+                            FinampSettingsHelper.finampSettings.playbackSpeed;
+                        _textController.text = currentSpeed.toString();
+                      });
+                    })),
+            Padding(
+              padding: EdgeInsets.only(top: 8.0, left: 15.0, right: 15.0),
+              child: Form(
+                key: _formKey,
+                child: Row(
+                  children: [
+                    SimpleButton(
+                      text: AppLocalizations.of(context)!.reset,
+                      icon: TablerIcons.arrow_back_up_double,
+                      iconColor: widget.iconColor,
+                      onPressed: () {
+                        _queueService.setPlaybackSpeed(1.0);
+                        setState(() {
+                          currentSpeed = 1.0;
+                          _textController.text = currentSpeed.toString();
+                        });
+                      },
+                    ),
+                    Expanded(
+                      child: Container(
+                        margin: EdgeInsets.symmetric(horizontal: 20.0),
```

Comment:
a little too big for my taste, and I prefer to use `paddings` instead. They increase the tapable area of widgets.